plugin/trino-exasol/src/test/java/io/trino/plugin/exasol/TestExasolConnectorTest.java

```diff
@@ -0,0 +1,470 @@
+                    SUPPORTS_AGGREGATION_PUSHDOWN,
+                    SUPPORTS_LIMIT_PUSHDOWN -> false;
+
+            // Parallel writing is not supported due to restrictions of the Exasol JDBC driver.
+            case SUPPORTS_CREATE_SCHEMA,
+                    SUPPORTS_CREATE_TABLE,
+                    SUPPORTS_RENAME_TABLE,
+                    SUPPORTS_INSERT,
+                    SUPPORTS_UPDATE,
+                    SUPPORTS_DELETE,
+                    SUPPORTS_ADD_COLUMN,
+                    SUPPORTS_RENAME_COLUMN,
+                    SUPPORTS_COMMENT_ON_TABLE,
+                    SUPPORTS_SET_COLUMN_TYPE -> false;
+
+            case SUPPORTS_ROW_TYPE,
+                    SUPPORTS_ARRAY,
+                    SUPPORTS_NEGATIVE_DATE // min date is 0001-01-01
+                    -> false;
+
+            default -> super.hasBehavior(connectorBehavior);
+        };
+    }
+
+    @Override
+    protected Optional<DataMappingTestSetup> filterDataMappingSmokeTestData(DataMappingTestSetup setup)
+    {
+        String type = setup.getTrinoTypeName();
+        if (type.equals("time") || type.startsWith("time(")) {
+            // Exasol does not have a TIME type
+            return Optional.of(setup.asUnsupported());
+        }
+        if (type.startsWith("timestamp") && type.endsWith("with time zone")) {
+            // Exasol only supports TIMESTAMP WITH _LOCAL_ TIMEZONE
+            return Optional.of(setup.asUnsupported());
+        }
+        if (type.startsWith("timestamp(6)")) {
+            // Exasol supports timestamps with milliseconds precision, not more.
+            return Optional.of(setup.asUnsupported());
+        }
+        if (type.startsWith("array") || type.startsWith("row")) {
+            // Exasol does not support composite types
+            return Optional.of(setup.asUnsupported());
+        }
+        if (setup.getTrinoTypeName().equals("char(3)") && setup.getSampleValueLiteral().equals("'ab'")) {
+            // Exasol fills char(3) with spaces, causing test to fail
+            return Optional.of(new DataMappingTestSetup("char(3)", "'abc'", "'zzz'"));
+        }
+        return Optional.of(setup);
+    }
+
+    @Override
+    protected TestTable createTableWithDefaultColumns()
+    {
+        return new TestTable(
+                onRemoteDatabase(),
+                TestingExasolServer.TEST_SCHEMA + ".test_default_cols",
+                "(col_required decimal(20,0) NOT NULL," +
+                        "col_nullable decimal(20,0)," +
+                        "col_default decimal(20,0) DEFAULT 43," +
+                        "col_nonnull_default decimal(20,0) DEFAULT 42 NOT NULL ," +
+                        "col_required2 decimal(20,0) NOT NULL)");
+    }
+
+    // Override test because INSERT is not supported
+    @Override
+    @Test
+    public void testNativeQuerySelectUnsupportedType()
+    {
+        try (TestTable testTable = createTableWithUnsupportedColumn()) {
```

Comment:
let's assert `hasBehaviour(SUPPORTS_INSERT)` returns false so that this test is updated when INSERTS are supported in future.